modules/tensor_mechanics/test/tests/viscoplasticity_stress_update/exact.i

```diff
@@ -0,0 +1,348 @@
+  [../]
+  [./M_onepointfive]
+    type = ParsedFunction
+    vars = 'avg_hydro onepointfive_gauge_stress'
+    vals = 'avg_hydro onepointfive_gauge_stress'
+    value = 'abs(avg_hydro) / onepointfive_gauge_stress'
+  [../]
+  [./Q_one]
+    type = ParsedFunction
+    vars = 'avg_vonmises one_gauge_stress'
+    vals = 'avg_vonmises one_gauge_stress'
+    value = 'avg_vonmises / one_gauge_stress'
+  [../]
+  [./M_one]
+    type = ParsedFunction
+    vars = 'avg_hydro one_gauge_stress'
+    vals = 'avg_hydro one_gauge_stress'
+    value = 'abs(avg_hydro) / one_gauge_stress'
+  [../]
+[]
+
+[Materials]
+  [./elasticity_tensor]
+    type = ComputeIsotropicElasticityTensor
+    youngs_modulus = 1e10
+    poissons_ratio = 0.3
+  [../]
+  [./strain]
+    type = ComputeMultiplePorousInelasticStress
+    inelastic_models = 'gtn lps_ten lps_five lps_three lps_two lps_onepointfive lps_one'
+    initial_porosity = 1e-3
+    outputs = all
+  [../]
+
+  [./gtn]
+    type = ViscoplasticityStressUpdate
+    coefficient = 0
+    power = 1 # arbitrary
+    viscoplasticity_model = GTN
+    base_name = gtn
+    outputs = all
+    relative_tolerance = 1e-30
+  [../]
+  [./lps_ten]
+    type = ViscoplasticityStressUpdate
+    coefficient = 0
+    power = 10
+    base_name = ten
+    outputs = all
+    relative_tolerance = 1e-30
+  [../]
+  [./lps_five]
+    type = ViscoplasticityStressUpdate
+    coefficient = 0
+    power = 5
+    base_name = five
+    outputs = all
+    relative_tolerance = 1e-30
+  [../]
+  [./lps_three]
+    type = ViscoplasticityStressUpdate
+    coefficient = 0
+    power = 3
+    base_name = three
+    outputs = all
+    relative_tolerance = 1e-30
+  [../]
+  [./lps_two]
+    type = ViscoplasticityStressUpdate
+    coefficient = 0
+    power = 2
+    base_name = two
+    outputs = all
+    relative_tolerance = 1e-30
+  [../]
+  [./lps_onepointfive]
+    type = ViscoplasticityStressUpdate
+    coefficient = 0
+    power = 1.5
+    base_name = onepointfive
+    outputs = all
+    relative_tolerance = 1e-30
+    # constant_on = ELEMENT
+  [../]
+  [./lps_one]
+    type = ViscoplasticityStressUpdate
+    coefficient = 0
+    power = 1
+    base_name = one
+    outputs = all
+    relative_tolerance = 1e-30
+    # constant_on = ELEMENT
+  [../]
+
+  [./const_stress]
+    type = ComputeExtraStressConstant
+    extra_stress_tensor = '1 1 1 1 1 1 1 1 1'
+  [../]
+[]
+
+[BCs]
+  [./no_disp_x]
+    type = PresetBC
+    variable = disp_x
+    boundary = left
+    value = 0.0
+  [../]
+  [./no_disp_y]
+    type = PresetBC
+    variable = disp_y
+    boundary = bottom
+    value = 0.0
+  [../]
+  [./no_disp_z]
+    type = PresetBC
+    variable = disp_z
+    boundary = back
+    value = 0.0
+  [../]
+  [./Pressure]
+    [./asdf]
+      boundary = 'top right front'
+      function = '10^(t/4.5)'
+    [../]
+  [../]
+[]
+
+[Preconditioning]
+  [./smp]
+    type = SMP
+    full = true
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = FD
```

Comment:
Here's an example @dschwen @lindsayad of AD kicking ass...This input won't even solve without FD without AD since I have a poor jacobian (i.e. no tangent operator). The tangent operator is something I worked big time on for power law creep last year, and sunk in many hours. I'm not even sure how to derive it for this case. Instead with this, I dropped it into AD, and it solves better than FD, and will scale waaay better. I think I'll remove all the non-AD implementation and just go full send into AD